vid/src/advz.rs

```diff
@@ -858,16 +882,35 @@ where
         DenseUVPolynomial::from_coefficients_vec(coeffs_vec)
     }
 
-    fn polynomial<I>(&self, coeffs: I) -> KzgPolynomial<E>
+    fn polynomial<I>(&self, coeffs: I, chunk_size: usize) -> KzgPolynomial<E>
     where
         I: Iterator,
         I::Item: Borrow<KzgEval<E>>,
     {
-        Self::polynomial_internal(
-            &self.eval_domain,
-            (self.recovery_threshold * self.multiplicity) as usize,
-            coeffs,
-        )
+        Self::polynomial_internal(&self.eval_domain, chunk_size, coeffs)
+    }
+
+    fn min_multiplicity(&self, payload_byte_len: u32, multiplicity: u32) -> u32 {
+        let elem_bytes_len =
+            bytes_to_field::elem_byte_capacity::<<E as Pairing>::ScalarField>() as u32;
+        let elems = payload_byte_len / elem_bytes_len;
+        let recovery_threshold = self.recovery_threshold;
+        if recovery_threshold * multiplicity < elems {
+            // payload is large. no change in multiplicity needed.
+            return multiplicity;
+        }
+
+        // payload is small: choose m such that 0 < m < multiplicity
+        // and m is the power of two that fits the payload best.
+        let mut m = multiplicity;
+        loop {
```

Comment:
ok I know this loop has only logarithmic complexity but I'm pretty sure we can replace it with a call to `ilog2` or something.

Reply:
This loop has a bug: the `if` condition is always met so the loop it always breaks immediately with `m = multiplicity`. Fixed in https://github.com/EspressoSystems/jellyfish/pull/667/commits/17d55fba72b2cb4739fb417525c57ac41e9523e0.

Reply:
Yes, you are absolutely right. Thanks for spotting this. This should have been `m=1` and then left-shift (`m<<1`) but your approach is definitely superior and more readable, imo.